docs/src/components/ExternalExampleLayout.tsx

```diff
@@ -49,34 +53,53 @@ class ExternalExampleLayout extends React.Component<
     })
 
     if (!examplePath) return <PageNotFound />
+
     const exampleSource: ExampleSource = exampleSourcesContext(examplePath)
-    const isRtlEnabled = rtl === 'true'
+    const theme = this.getTheme()
 
     return (
-      <Provider key={this.state.renderId} theme={{ rtl: isRtlEnabled }}>
-        <div dir={isRtlEnabled ? 'rtl' : undefined}>
-          <SourceRender
-            babelConfig={babelConfig}
-            source={exampleSource.js}
-            renderHtml={false}
-            resolver={importResolver}
-          >
-            <SourceRender.Consumer>
-              {({ element, error }) => (
-                <>
-                  {element}
-                  {/* This block allows to see issues with examples as visual regressions. */}
-                  {error && (
-                    <div style={{ fontSize: '5rem', color: 'red' }}>{error.toString()}</div>
+      <Provider key={this.state.renderId} theme={theme}>
+        <Provider.Consumer
```

Comment:
this approach as taken from `docs/src/components/ComponentDoc/ComponentExample/ComponentExample.tsx`, line 588:
```jsx
<Provider.Consumer
  render={({ siteVariables }) => {
    return (
      <Segment
        dir={showRtl ? 'rtl' : undefined}
        className={`rendered-example ${this.getKebabExamplePath()}`}
        styles={{
          padding: '2rem',
          color: siteVariables.bodyColor,
          backgroundColor: siteVariables.bodyBackground,
          ...(showTransparent && {
            backgroundImage:
              'url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAoAAAAKCAYAAACNMs+9AAAAKUlEQVQoU2NkYGAwZkAD////RxdiYBwKCv///4/hGUZGkNNRAeMQUAgAtxof+nLDzyUAAAAASUVORK5CYII=")',
            backgroundRepeat: 'repeat',
          }),
        }}
      >
        <SourceRender.Consumer>{({ element }) => element}</SourceRender.Consumer>
      </Segment>
    )
  }}
/>
```

 I think we should have the same approach in both cases (use segment with `2rem` padding) and extract logic in separate component. I'll probably do that, what do you think?

Reply:
Let's leave as is, because #852 will apply styles and we will not need it at all 🌟

Reply:
oh, that answers my question above..hmmm the tests will be useless until then, I don't really like that especially that the #852 is not that active anymore

Reply:
agreed to split this in 2 PRs to avoid screen regressions here, the new PR will depend on #852

Reply:
resolving, tracked in newly opened #1050